### DISCUSSION TITLE
How to  make zfs receive  property likes refreservation the same as source volume?

### DISCUSSION THREAD

**Discussion by guanfeix** [openzfs/zfs#14022] [Q&A]
I create two volumes which both are thick volume, but receive volume turn into thin volume after receive operation
this is my cmd:
>zfs send -p -P -v -R  xx/yy@zz | zfs recv -v -s -F  pool/receive_volume

other property  in receive_volume like compression  /dedup just the same as the source volume.But the refreservation is rollback to None(the same as pool)
![image](https://user-images.githubusercontent.com/34960209/195286833-c6c101d3-1900-436b-bf07-bb48f12c6153.png)
Any buddys know How to make the property likes refreservation of volume which run  "zfs receive" just same as source volume

**Comment by guanfeix** [comment]
Also can not force set  refreservation=auto, in zfs receive -o  option,which report size is greater than available space.
![image](https://user-images.githubusercontent.com/34960209/195301598-7abeaa6a-f27b-40f2-b05d-9926ffca78b1.png)

**Comment by rincebrain** [comment]
That's odd, I'd rather expect it to have failed to receive if it couldn't set the refreservation correctly.

`zfs inherit -S` should, I think, make it change the value back to the one it logged as being received, though that might error for the same reason as the initial receive set did.

Based on the values in your other question, though, that error does make sense - if the zvols are 5.2T each or so, even if they're only using 500G or so, refreservation is going to reserve enough space to overwrite the whole volume once to avoid you being able to run out of space mid-overwrite, since almost nothing knows what to do about a block device that returns ENOSPC, and would be 5.2T, which is more space left after other reservations than the pool has available...

**Reply by guanfeix** [reply]
It's a different environment. the volume which I test only 10~20G, the pool which has 20T space. I just want to know how to behaves in normal case. If we can keep refreservation  consistent after "zfs  receive" operation.
![image](https://user-images.githubusercontent.com/34960209/195490786-962eb6a0-c8d1-4d86-a47a-4af925fb1482.png)

when I change an environment. from 2.0.7 to  0.8.6,it still the same which other properties exclude the refreservation were transmited even the output change,just seems more detailed
![image](https://user-images.githubusercontent.com/34960209/195494610-f45cd43b-1256-49b5-b655-13c6394d8abb.png)
